src/foam/nanos/http/AuthWebAgent.java

```diff
@@ -267,22 +267,10 @@ public Session authenticate(X x) {
         // user should not be null, any login failure should throw an Exception
         logger.error("AuthService.login returned null user and did not throw AuthenticationException.");
         // TODO: generate stack trace.
-        if ( ! SafetyUtil.isEmpty(authHeader) ) {
-          sendError(x, resp, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Authentication failure.");
-        } else {
-          if ( ! getRedirectToLogin(x) ) {
-            PrintWriter out = x.get(PrintWriter.class);
-            out.println("Authentication failure.");
-          }
-        }
```

Comment:
is this ok? i dont see how we can ever reach this code without auth header so i send error right away. Also we do suspicious number of checks for empty auth header. Shouldn't empty header be treated in a very similar way as a invalid auth header?

Reply:
I think the idea here is to handle the case in the DIG UI. Can you test out the DIG UI to see what happens when you send the request from the button on that page?

Reply:
yeah, the button works but you cant follow url anymore(because plain GET url not including any auth). 
From discussion with Kevin on slack:
```The reason is because it used to have a simple non-FOAM GUI in case a user accessed it directly, to help them build their query.

But that became too laborious so it was replaced by a proper FOAM UI, so that support is no longer needed.```